controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -1591,12 +1592,11 @@ private HttpResponse testConfig(ApplicationId id, JobType type) {
                                     .collect(Collectors.toSet());
         var testedZone = type.zone(controller.system());
 
-        // This should not happen, unless one asks for test-config for a production job, which one shouldn't have use for.
-        if (deployments.stream().anyMatch(deploymentId ->      deploymentId.zoneId().equals(testedZone)
-                                                          && ! deploymentId.applicationId().equals(id)))
-            throw new IllegalStateException("Conflict: " + testedZone + " contains both default and " + id.instance().value() + " instances");
+        // If a production job is specified, the production deployment of the _default instance_ is the relevant one,
+        // as user instances should not exist in prod. TODO jonmv: Remove this when multiple instances are supported (above).
+        if ( ! type.isProduction())
+            deployments.add(new DeploymentId(id, testedZone));
```

Comment:
Collectors.toSet() makes no guarantee about the mutability of the returned Set: "There are no guarantees on the type, mutability, ...". Because of this I have never tried nor seen a `.add` so I guess it works... But consider using the toSet(Supplier) instead.

Reply:
Will do.